\documentclass{article}

% Bibliography
\usepackage{natbib}
\bibpunct{(}{)}{;}{a}{}{;}

% Use 'It was found that A is B (Name 1234)' style
\setcitestyle{authoryear,open={},close={}}

% Affiliations
\usepackage{authblk}
\title{pirouette: the error BEAST2 makes in inferring a phylogeny}
\author[1]{Rich\`el J.C. Bilderbeek}
\author[1]{Giovanni Laudanno}
\author[1]{Rampal S. Etienne}
\affil[1]{Groningen Institute for Evolutionary Life Sciences, University of Groningen, Groningen, The Netherlands}

% Use double spacing
\usepackage{setspace}
\doublespacing

\usepackage{listings}
\usepackage{hyperref}
\usepackage{todonotes}
\usepackage{verbatim}
\usepackage{pgf}
\usepackage{bm}

% sidewaysfigure
\usepackage{rotating}

% Style of listings
% From http://r.789695.n4.nabble.com/How-to-nicely-display-R-code-with-the-LaTeX-package-listings-tp4648110.html
\usepackage{fancyvrb} 
\definecolor{codegreen}{rgb}{0,0.6,0}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolor}{rgb}{0.95,0.95,0.92}
\lstdefinestyle{mystyle}{
  language=R,% set programming language
  basicstyle=\ttfamily\small,% basic font style
  commentstyle=\color{gray},% comment style
  % numbers=left,% display line numbers on the left side
  numberstyle=\scriptsize,% use small line numbers
  numbersep=10pt,% space between line numbers and code
  tabsize=2,% sizes of tabs
  showstringspaces=false,% do not replace spaces in strings by a certain character
  captionpos=b,% positioning of the caption below
  breaklines=true,% automatic line breaking
  escapeinside={(*}{*)},% escaping to LaTeX
  fancyvrb=true,% verbatim code is typset by listings
  extendedchars=false,% prohibit extended chars (chars of codes 128--255)
  alsoletter={.<-},% becomes a letter
  alsoother={$},% becomes other
  otherkeywords={!=, ~, $, \&, \%/\%, \%*\%, \%\%, <-, <<-, /},% other keywords
  deletekeywords={c}% remove keywords 
}
\lstset{style=mystyle}

% Adds numbered lines
\usepackage{lineno}
\linenumbers

% Rename 'Abstract' to 'Summary 
\usepackage[english]{babel}
\addto{\captionsenglish}{\renewcommand{\abstractname}{Summary}}

%comments
\newcommand{\giovanni}[1]{\textcolor{blue}{\textbf{[GL: #1]}}}

\begin{document}

\maketitle

\begin{abstract}

  \textbf{1. }
    BEAST2 is a popular Bayesian phylogenetics software tool,
    that takes an alignment and inference model to create a
    posterior of jointly-estimated phylogenies and model parameter estimates.
    When a new macro-evolutionary speciation \giovanni{maybe "diversification" is a better word} model is developed,
    a good first step is to measure the error BEAST2 makes on a known
    phylogeny derived from the new speciation process \giovanni{I'd use "diversification mechanisms" in place of "speciation process"}, 
    with its current set of inference models. \\
  \textbf{2. }
    Here, we present a free, libre and open-source R package, \verb;pirouette;
    that assesses the inference error BEAST2 makes based on a known/true 
    phylogeny. \\
  \textbf{3. }
    We describe \verb;pirouette;'s usage and the biological scientific
    question it can answer, including full examples. \\
  \textbf{4. }
    As \verb;pirouette; is designed to be of high quality and extendable, 
    we conclude by describing the further development of the package. \\
\end{abstract}

{\bf Keywords:} computational biology, evolution, phylogenetics, BEAST2, pirouette, R





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Phylogenies are commonly used to explore evolutionary hypotheses.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Description}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\verb;pirouette; is written in the R programming language (\cite{R}).

\begin{sidewaysfigure}
  \centering
  \includegraphics[width=\textwidth]{overview.png}
  \caption{pirouette pipeline}
  \label{fig:pipeline}
\end{sidewaysfigure}

The goal of \verb;pirouette; is to measure the inference error BEAST2
makes from a given/known/true phylogeny \giovanni{I'd use: "given phylogeny of which full information is provided"}. This involves multiple steps,
described in detail below.


\giovanni{suggestion: expressing the same concepts of the "in brief" section in points could be an easier experience for the reader}
\giovanni{start example}


The pipeline is summarized by the following steps:
\begin{enumerate}
    \item from a given phylogeny an alignment is simulated;
    \item the alignment is used as BEAST2 input to infer a posterior distribution of phylogenies;
    \item posterior phylogenies are compared with the given phylogenies to estimate the error we make. The bigger the match, the lower the error;
\end{enumerate}
There is also the option to generate a 'twin tree',
that goes through the same pipeline.
\giovanni{end example}


In brief: from a given phylogeny, an alignment is simulated, from which a BEAST2
posterior is inferred, of which its phylogenies are compared with the given
phylogeny. The more the posterior phylogenies match the given phylogeny,
the lower the error. There is also the option to generate a 'twin tree',
that goes through the same pipeline.

The first step simulates a DNA alignment from the given phylogeny.
One can specify a DNA sequence
of any length at the root of the phylogeny, a DNA mutation rate, a
site (also: \giovanni{I'd use "i.e."} nucleotide substitution) model, 
a clock model, a random number generator (RNG) seed and a location
where the alignment is saved to. This step is relatively fast, but longer
DNA alignments will noticeably slow down the inference step.

The second step selects an inference model, using the alignment.
The user can specify \giovanni{whether} to use the generative model or select a best model
from a set of inference models. 
When selecting the generative model,
the site and clock model used in the alignment simulation are used
in \giovanni{the} inference. Because the given phylogeny (on which the alignment is based)
may have followed any tree prior (also: \giovanni{I'd use "i.e."} speciation model), the user needs
to specify which tree prior is used in \giovanni{the} inference. 
When selecting a \giovanni{I'd use "the"} best
model, the alignment is used to find the inference model that has the
highest evidence (also: \giovanni{I'd use "i.e."} marginal likelihood) from a set of inference models.
The evidence of an inference model is estimated using a nested sampling (NS)
approach, as described in \cite{maturana2018model}. The nested sampling is
performed by \verb;babette;, that call the 'NS' BEAST2 package. 
Using BEAST2 packages (in a scripted way) can only be done under Linux and Mac; inference model selection is not available on Windows.

The third step infers a Bayesian posterior from the simulated alignment,
using the inference model(s) selected in the previous step. The user
can specify the additional parameters needed for the BEAST2 run, which
are the Markov-Chain Monte Carlo (MCMC) setup, 
an optional Most Recent Common Ancestor (MRCA) prior and an RNG seed.
The MCMC setup determines the number of posterior trees sampled.
An MRCA prior allows the inferred phylogeny to have a dated crown age.

The fourth step measures the inference error, using the phylogenies in the
Bayesian posterior. These phylogenies are compared to the given
phylogeny using an error statistic, which is the nLTT 
statistic \cite{janzen2015approximate}. Additionally, the user can specify the
proportion of posterior phylogenies to discard (also called: the burn-in \giovanni{as far as I know the burn-in is to discard the start of a chain because it's still too far from the convergence, see http://www.beast2.org/glossary/}).

An optional step is to generate a 'twin tree', that will be
analyzed in the same way as the true tree. The twin tree has the same
topology as the given phylogeny, yet its branch lengths follow the same
distribution as a Yule (pure-birth) or (constant rate) birth-death tree prior.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Installation}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\verb;pirouette; can be installed easily from CRAN:

\begin{lstlisting}[language=R, floatplacement=H]
install.packages("pirouette")
\end{lstlisting}

For the most up-to-date version, 
one can download and install the package from \verb;pirouette;'s GitHub repository:

\begin{lstlisting}[language=R, floatplacement=H]
usthis::install_github("richelbilderbeek/pirouette")
\end{lstlisting}
\giovanni{is "usthis" supposed to be "devtools"?}
To start using \verb;pirouette;, load its functions in the global namespace first:

\begin{lstlisting}[language=R, floatplacement=H]
library(pirouette)
\end{lstlisting}
Because \verb;pirouette; calls BEAST2, BEAST2 must be installed. 
This can be done from within R, using:

\begin{lstlisting}[language=R, floatplacement=H]
install_beast2()
\end{lstlisting}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Usage: first research question}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

A first research question that \verb;pirouette; answers is:
What is the error BEAST2 makes from a phylogeny using the same 
speciation \giovanni{I'd use: "diversification"} model as it was generated by?

We start with an idealised birth-death tree of five taxa and a crown age of 10:

% Need idealised birth-death tree
% From https://github.com/richelbilderbeek/pirouette_article/issues/1
% and https://github.com/richelbilderbeek/pirouette/issues/35
\begin{lstlisting}[language=R, floatplacement=H]
phylogeny <- ape::read.tree(text = "((A:4, B:4):1, (C:4, D:4):1);")
\end{lstlisting}

The first step in \verb;pirouette; is to simulate a DNA alignment from the 
given phylogeny. To do so, we must specify the DNA root sequence
and a mutation rate. In this example, the DNA root sequence consists
out of four block of 250 nucleotides each, where the per-nucleotide
mutation rate is on average \giovanni{I don't think it's correct to say that's "on average". Usually when you have a rate $\lambda$ you suppose to use a Poisson process generated by this rate: $\frac{\lambda^k e^\lambda}{k!}$. $\lambda$ turns out to be also the expected number of occurrences} 0.1 mutations per unit time:

\begin{lstlisting}[language=R, floatplacement=H]
alignment_params <- create_alignment_params(
  root_sequence = create_blocked_dna(length = 1000),
  mutation_rate = 0.1
)
\end{lstlisting}

The second step is to select how an inference model is picked.
We just select the generative model, which consists in a collection of three elements: a site model, a clock
model of the alignment and a birth-death tree prior
to be the generative tree prior.

% Note that this syntax does not work yet, see
% https://github.com/richelbilderbeek/pirouette/issues/36
\begin{lstlisting}[language=R, floatplacement=H]
model_select_param <- create_gen_model_select_param(
  alignment_params = alignment_params,
  tree_prior = create_bd_tree_prior()
)
\end{lstlisting}

The third and fourth step have sensible defaults, and we are not
interested in using a twin tree. We can now measure the errors BEAST2
makes when inferring the given phylogeny in this way:

\begin{lstlisting}[language=R, floatplacement=H]
errors <- pir_run(
  phylogeny,
  alignment_params = alignment_params,
  model_select_params = model_select_param
)
\end{lstlisting}

\verb;pirouette; has a nice plotting function:

\begin{lstlisting}[language=R, floatplacement=H]
pir_plot(errors)
\end{lstlisting}

% TODO: write script to create figure 2, see
% https://github.com/richelbilderbeek/pirouette/issues/37
The resulting figure is shown in figure 2.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Usage: second research question}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

A second research question that \verb;pirouette; answers, is:
What is the error BEAST2 makes from a phylogeny when
picking the best inference model?

Here we start with a tree \giovanni{generated} from an unknown speciation \giovanni{I'd use: "diversification"} model,
that has four taxa and a crown age of 5:

% Need idealised birth-death tree
% From https://github.com/richelbilderbeek/pirouette_article/issues/1
% and https://github.com/richelbilderbeek/pirouette/issues/35
\begin{lstlisting}[language=R, floatplacement=H]
phylogeny <- ape::read.tree(text = "((A:4, B:4):1, (C:4, D:4):1);")
\end{lstlisting}

The first step in \verb;pirouette; is to simulate a DNA alignment from the 
given phylogeny. We'll use the same parameters here:

\begin{lstlisting}[language=R, floatplacement=H]
alignment_params <- create_alignment_params(
  root_sequence = create_blocked_dna(length = 1000),
  mutation_rate = 0.1
)
\end{lstlisting}

The second step is to select how an inference model is picked.
We will let the inference model that has the highest evidence be used
in the Bayesian inference. By default, \verb;pirouette; will try
all combinations of site models, clock models and tree priors.

% Note that this syntax does not work yet, see
% https://github.com/richelbilderbeek/pirouette/issues/36
\begin{lstlisting}[language=R, floatplacement=H]
model_select_param <- create_best_model_select_param()
\end{lstlisting}

Also here, the third and fourth step have sensible defaults, and we are not
interested in using a twin tree. We can now measure the errors BEAST2
makes when inferring the given phylogeny like this:

\begin{lstlisting}[language=R, floatplacement=H]
errors <- pir_run(
  phylogeny,
  alignment_params = alignment_params,
  model_select_params = model_select_param
)
\end{lstlisting}

Again, showing the results:

\begin{lstlisting}[language=R, floatplacement=H]
pir_plot(errors)
\end{lstlisting}

% TODO: write script to create figure 3, see
% https://github.com/richelbilderbeek/pirouette/issues/38
The resulting figure is shown in figure 3.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Usage: third research question}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\giovanni{I noticed that "phylogeny", "alignment params" and "model select param" are needed for all the three research questions. Some times, though, you repeat yourself. In an article you want to be as concise as possible so I'd suggest to avoid repetitions. One solution could be to summarize all the parameter settings in a summary scheme}

A third research question that \verb;pirouette; answers is:
What is the error BEAST2 makes from a phylogeny 
picking the best inference model, compared to the background noise?

Here we start again with a tree from an unknown speciation \giovanni{I'd use: "diversification"} model,
that has four taxa and a crown age of 5:

% Need idealised birth-death tree
% From https://github.com/richelbilderbeek/pirouette_article/issues/1
% and https://github.com/richelbilderbeek/pirouette/issues/35
\begin{lstlisting}[language=R, floatplacement=H]
phylogeny <- ape::read.tree(text = "((A:4, B:4):1, (C:4, D:4):1);")
\end{lstlisting}

The first step in \verb;pirouette; is to simulate a DNA alignment from the 
given phylogeny. We'll use the same parameters here:

\begin{lstlisting}[language=R, floatplacement=H]
alignment_params <- create_alignment_params(
  root_sequence = create_blocked_dna(length = 1000),
  mutation_rate = 0.1
)
\end{lstlisting}

The second step is to select how an inference model is picked.
We will let the inference model that has the highest evidence be used
in the Bayesian inference. By default, \verb;pirouette; will try
all combinations of site models, clock models and tree priors.

% Note that this syntax does not work yet, see
% https://github.com/richelbilderbeek/pirouette/issues/36
\begin{lstlisting}[language=R, floatplacement=H]
model_select_param <- create_best_model_select_param()
\end{lstlisting}

Also here, the third and fourth step have sensible defaults. 

This time, we are interested in creating a twin tree. A twin tree
has the same topology as the given tree, yet its branch lengths follow
a birth-death distribution. Creating a twinning parameter is easy,
as it has sensible default settings:

\begin{lstlisting}[language=R, floatplacement=H]
twinning_params <- create_twinning_params()
\end{lstlisting}


We can now measure the errors BEAST2
makes when inferring the given phylogeny, compared
to the background noise, like this:

\begin{lstlisting}[language=R, floatplacement=H]
errors <- pir_run(
  phylogeny,
  twinning_params = twinning_params,
  alignment_params = alignment_params,
  model_select_params = model_select_param
)
\end{lstlisting}

Again, showing the results:

\begin{lstlisting}[language=R, floatplacement=H]
pir_plot(errors)
\end{lstlisting}

% TODO: write script to create figure 4, see
% https://github.com/richelbilderbeek/pirouette/issues/39
The resulting figure is shown in figure 4
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{pirouette resources}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\verb;pirouette; is free, libre and open source software available at 
\url{http://github.com/richelbilderbeek/pirouette}
and is licensed under the GNU General Public License v3.0.
\verb;pirouette; uses the Travis CI (\url{https://travis-ci.org})
continuous integration service, which is known to significantly 
increase the number of bugs exposed (\cite{vasilescu2015}) and increases
the speed at which new features are added (\cite{vasilescu2015}).
\verb;pirouette; has a 100\% code coverage, which correlates with 
code quality (\cite{horgan1994,del1995correlation}). 
\verb;pirouette; follows Hadley Wickham's style guide (\cite{style_guide}), 
which improves software quality (\cite{fang2001}).
\verb;pirouette; depends on multiple packages, which are 
\verb;ape; (\cite{APE}), 
\verb;babette; (\cite{bilderbeek2018babette}),
\verb;ggplot2; (\cite{ggplot2}),
\verb;knitr; (\cite{knitr}),
\verb;mcbette; (\cite{mcbette}),
\verb;phangorn; (\cite{phangorn}),
\verb;rmarkdown; (\cite{rmarkdown}),
\verb;stringr; (\cite{stringr}),
\verb;testit; (\cite{testit}) and 
\verb;usethis; (\cite{usethis}).

\verb;pirouette;'s development takes place on GitHub,
\url{https://github.com/richelbilderbeek/pirouette}, 
which accommodates collaboration (\cite{perez2016ten}) 
and improves transparency (\cite{gorgolewski2016practical}).
\verb;pirouette;'s GitHub facilitates feature requests and 
has guidelines on how to do so.

\verb;pirouette;'s documentation is extensive. All functions are documented
in the package's internal documentation. For quick use, 
each exported function shows a minimal example. 
For easy exploration, each exported function's documentation links to related functions.
Additionally, \verb;pirouette; has a vignette that demonstrates extensively how
to use it. There is documentation on the GitHub to get started, 
with a dozen examples of BEAUti screenshots with equivalent \verb;pirouette; code.
Finally, \verb;pirouette; has tutorial videos that can 
be downloaded or viewed on YouTube, \url{https://goo.gl/weKaaU} \giovanni{be sure that shortened link are not deprecated after a certain time}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Citation of pirouette}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Scientists using \verb;pirouette; in a published paper can cite this
article, and/or cite the \verb;pirouette; package 
directly. To obtain this citation from within an R script, use:

\begin{lstlisting}[language=R]
> citation("pirouette")
\end{lstlisting}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Acknowledgements}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

We would like to thank the Center for Information Technology of the University 
of Groningen for their support and for providing access to the Peregrine 
high performance computing cluster. 
We thank the Netherlands 
Organization for Scientific Research (NWO) for financial support 
through a VICI grant awarded to RSE.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Data Accessibility}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

All code is archived at \url{http://github.com/richelbilderbeek/pirouette_article},
with DOI \url{https://doi.org/12.3456/zenodo.1234567}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Authors' contributions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

RJCB, GL and RSE conceived the idea for the package. 
RJCB created and tested the package, and wrote the first draft of the manuscript.
GL tested the package and contributed substantially to revisions.
RSE contributed to revisions.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Bibliography
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% MEE style
\bibliographystyle{mee}
\bibliography{article}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\end{document}
