%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Supplementary material}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

This supplementary material contains additional facets of \verb;pirouette;, 
such as the installation of the package, an overview of
pirouette's main functions and a guide for users, based on multiple experiments
that are shown here as well.

For these experiments, we limited the number of replicates by time, 
aiming at a duration of
24 hours per setting, when run on the Peregrine computer cluster of the
University of Groningen. Due to this, for example, a run of 40 taxa only
has 6 replicates, because one run takes 4 hours. For all experiments, the intermediate results can all be downloaded 
from their respective websites, which is approximately 5 gigabyte in total.

All the figures shown in this section are shown without any aesthetical modifications, with the exception that the arrangement
of the sub-figures in subsection \ref{subsec:main_example},
where we aligned parts of the figure by hand.

Here is an overview of the various sections:

\begin{itemize}
  \item{
    subsection \ref{subsec:guidelines}: guidelines for users
  }
  \item{
    subsection \ref{subsec:installation}: installation
  }
  \item{
    subsection \ref{subsec:resources}: resources, such as 
    website, tutorials, packages used, bug reporting and contributing
  }
  \item{
    subsection \ref{subsec:citation}: citation of pirouette
  }
  \item{
    subsection \ref{subsec:twinning}: the
    twinning process
  }
  \item{
    subsection \ref{subsec:candidates}: candidate models for the inference
  }
  \item{
    subsection \ref{subsec:stochasticity}: the effects of stochasticity
  }
  \item{
    subsection \ref{subsec:nltt}: the
    nLTT statistic
  }
  \item{
    subsection \ref{subsec:main_functions}: main functions
  }
  \item{
    subsection \ref{subsec:main_example}: code, extra figures and
    diagnostics regarding the main example.
  }
  \item{
    subsection \ref{subsec:distribution}: the result of using 
    multiple trees, as generated by the
    same stochastic process as the main example
  }
  \item{
    subsection \ref{subsec:n_taxa}: the effect of the number of taxa
  }
  \item{
    subsection \ref{subsec:n_nucleotides}: the effect of the DNA
    alignment sequence length
  }
  \item{
    subsection \ref{subsec:simplest_correct_parameterization} shows the
    effect when performing inference in the simplest use case
  }
  \item{
    subsection \ref{subsec:under_parameterization} shows the
    effect when performing inference with an under-parameterization
  }
  \item{
    subsection \ref{subsec:different_n_mutations} shows the
    effect when the twin alignment is allowed to have a different
    number of \new{substitution}s
  }
  \item{
    subsection \ref{subsec:mutation_rate} shows the
    effect of different mutation rates
  }
  \item{
    subsection \ref{subsec:acknowledgments}: Acknowledgments
  }
  \item{
    subsection \ref{subsec:data_accessibility}: Data accessibility
  }
  \item{
    subsection \ref{subsec:author_contributions}: Author contributions
  }
\end{itemize}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Guidelines for users}
\label{subsec:guidelines}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

From the experiments shown below, we composed some rough guidelines.
These guidelines should be treated as preliminary results, as
the total runtime of these experiments is 'only' 19 days.

\begin{itemize}
  \item{
    The use of 20 replicates results in decent plots.
  }
  \item{
    The use of more taxa increases the inference error
  }
  \item{
    The use of longer DNA sequences decreases the inference error.
  }
  \item{
    When we do not impose the same number of \new{substitution}s 
    between true and twin alignment, we observe a difference in the error 
    distributions with respect to the standard case (presented in the main 
    text) where they are forced to have the same number of \new{substitution}s.
  }
  \item{
    Using a mutation rate less than 1.0 / crown age, decreases the
    inference error. We predict this will increase the error in the parameter
    estimation.
  }
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Installation}
\label{subsec:installation}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\verb;pirouette; will be made available on CRAN from which 
it can then be easily installed:
\begin{lstlisting}[language=R, floatplacement=ht, frame=single]
install.packages("pirouette")
\end{lstlisting}

Until it is on CRAN, and for the most up-to-date version, 
one can download and install the package from \verb;pirouette;'s GitHub 
repository. We first need the \verb;mcbette; and \verb;nodeSub; packages:
\begin{lstlisting}[
    language = R,
    floatplacement = ht,
    frame = single
]
remotes::install_github(
  "richelbilderbeek/mcbette"
)
remotes::install_github(
  "thijsjanzen/nodeSub"
)
\end{lstlisting}

Now we can install \verb;pirouette;:
\begin{lstlisting}[
    language = R,
    floatplacement = ht,
    frame = single
]
remotes::install_github(
  "richelbilderbeek/pirouette"
)
\end{lstlisting}
which also installs its dependencies from CRAN.

To start using \verb;pirouette;, 
load its functions in the global namespace first:
\begin{lstlisting}[language=R, floatplacement=ht, frame=single]
library(pirouette)
\end{lstlisting}
Because \verb;pirouette; calls BEAST2, BEAST2 must be installed. 
This can be done from within R, using:
\begin{lstlisting}[language=R, floatplacement=ht, frame=single]
beastier::install_beast2()
\end{lstlisting}
For the option to select the best candidate model,
\verb;pirouette; needs the "NS" BEAST2 package [\cite{russel2019model}].
It can be installed from within R, using:
\begin{lstlisting}[language=R, floatplacement=ht, frame=single]
mauricer::install_beast2_pkg("NS")
\end{lstlisting}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Resources}
\label{subsec:resources}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\verb;pirouette; is free, libre and open source software available at 
\begin{sloppypar}
  \url{http://github.com/richelbilderbeek/pirouette},
\end{sloppypar}
licensed under the GNU General Public License version 3.
\verb;pirouette; depends on multiple packages, which are:
\verb;ape; [\cite{ape}],
\verb;assertive; [\cite{assertive}],
\verb;babette; [\cite{bilderbeek2018babette}],
\verb;DDD; [\cite{DDD}],
\verb;devtools; [\cite{devtools}],
\verb;dplyr; [\cite{dplyr}],
\verb;ggplot2; [\cite{ggplot2}],
\verb;knitr; [\cite{knitr}],
\verb;lintr; [\cite{lintr}],
\verb;magrittr; [\cite{magrittr}],
\verb;mcbette; [\cite{mcbette}],
\verb;nLTT; [\cite{nLTT}],
\verb;phangorn; [\cite{phangorn}],
\verb;phytools; [\cite{phytools}],
\verb;plyr; [\cite{plyr}],
\verb;rappdirs; [\cite{rappdirs}],
\verb;rmarkdown; [\cite{rmarkdown}],
\verb;Rmpfr; [\cite{Rmpfr}],
\verb;stringr; [\cite{stringr}],
\verb;TESS; [\cite{TESS,hohna2016tess}],
\verb;testit; [\cite{testit}], 
\verb;testthat; [\cite{testthat}] and
\verb;tidyr; [\cite{tidyr}].

\verb;pirouette;'s development takes place on GitHub,
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette},
\end{sloppypar}
which allows submitting bug reports, requesting features, 
and adding code. To improve quality, \verb;pirouette; 
uses a continuous integration service, has a code coverage of above 95\%
and enforces the most commonly used R style guide [\cite{style_guide}].

\verb;pirouette;'s is extensively documented on its website,
its documentation and its vignettes.
The \verb;pirouette; website is a good starting point to learn
how to use \verb;pirouette;, as it links to tutorials and videos.
The \verb;pirouette; package documentation describes
all functions and liberally links to related functions.
All exported functions show a minimal example as part of their documentation.
The \verb;pirouette; vignette demonstrates extensively how 
to use \verb;pirouette; in a more informally written way. 

The code used in this article and more examples that are periodically 
tested, can be found at
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_examples}. 
\end{sloppypar}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Citation of pirouette}
\label{subsec:citation}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

To cite \verb;pirouette; this article from within R, use:

\begin{lstlisting}[language=R]
> citation("pirouette")
\end{lstlisting}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The twinning process}
\label{subsec:twinning}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\verb;pirouette; allows to perform
a control measurement, by use of a process we call twinning. This control results in an error distribution
that is the baseline error of the pipeline. The difference
between the 'true' and 'twin' error distributions is caused only by the mismatch between the true tree model and the 
tree prior used in the actual inference.

The twinning process, $T$, encompasses two steps:
$T_1$, that generates a 'twin tree' (Fig.~\ref{fig:pipeline}, 1b) 
and $T_2$, which generates a 'twin alignment' (Fig.~\ref{fig:pipeline}, 2b).
Both twin tree and alignment will be analyzed in the same way as the true tree and alignment.

We define a phylogeny $\tau$ as the combination of
branching times $\Vec{t}$ and topology $\psi$, 
and denote as $\tau_{\mathit{G}}$ the phylogeny 
produced by a (possibly non-standard) generative diversification model, 
having branching times $\Vec{t}_{\mathit{G}}$ and 
topology $\psi_{\mathit{G}}$.

The first step ($T_1$) of the twinning process creates a tree $\tau_{\mathit{T}}$
with branching times $\Vec{t}_{\mathit{T}}$ while preserving the original
topology $\psi_{\mathit{G}}$:
\begin{align}
  \tau_{\mathit{G}} = (\Vec{t}_{\mathit{G}}, \psi_{\mathit{G}}) 
  \xrightarrow[]{\mathit{T_1}} 
  \tau_{\mathit{T}} = (\Vec{t}_{\mathit{T}}, \psi_{\mathit{G}})
\end{align}

\new{
  We chose to preserve the original topology to 
  increase the similarity between the twin to the original tree. 
  This works well in the cases of BD or DD models 
  we consider in our example, because all these models make the same
  assumption about topology (all topologies are equally likely).
  However, this might not be suitable for new models that assign different probabilities
  to trees with the same branching times but different topologies.
}
\new{
  The default option for the twin diversification model $p_T$ 
  is the standard BD model.
}
\verb;pirouette;
\new{
   has a built-in function to use a Yule model as well.
  Additionally, a user can specify a function to generate a twin tree
  from any speciation model, such as, for example, a coalescent model.
}

It is then possible to use the likelihood function 
$L_{\mathit{T}}$ for this diversification model to find 
the parameters $\theta^{*}_{\mathit{T}}$ 
(e.g. speciation and extinction rates, in case of a BD model) 
that maximize this likelihood applied to the true tree, conditioned on its number of tips $n_{\mathit{G}}$:
\begin{align}
    \max[L_{\mathit{T}}(\theta_{\mathit{T}}|\tau_{\mathit{G}}, n_{\mathit{G}})] 
\rightarrow \theta^{*}_{\mathit{T}}.
\end{align}
We use $\theta^{*}_{\mathit{T}}$ to simulate a number 
$n_{\mathit{T}} = n_{\mathit{G}}$ 
of branching times $\Vec{t}_{\mathit{T}}$ for the twin tree 
$\tau_{\mathit{T}}$, under the process $p_{T}$, 
while preserving the topology. 
We simulate the new branching times using the TESS 
package [\cite{TESS}].
\new{
  For simplicity, when simulating phylogenies we assumed a sampling fraction 
  of $100\%$. A different choice might have an effect on model performance.
}

The second step ($T_2$) of the twinning process simulates the twin alignment 
with the same clock model, site model and mutation rate 
used to simulate the alignment on the true. 
The twin alignment can be simulated in any user-defined way.
\verb;pirouette; provides the option simulate it with the same mutation rate as the true alignment. By default, however,
not only the same mutation rate is used, but also the total number of 
\new{substitution}s
matches the true alignment. The total number of \new{substitution}s is defined
as the number of different nucleotides between the (known) root sequence
compared to the sequences at the tips.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Candidate models}
\label{subsec:candidates}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The user has to specify exactly one standard inference model,
but may be unsure which one to pick. To account for this, the user can
specify a set of candidate inference models. Each of these candidate inference models is run in an initial, relatively short, analysis; the candidate model with the highest 
evidence (i.e., marginal likelihood) will then be
used in another, longer, inference run, resulting in another error distribution.
The evidence for an inference model is estimated by nested 
sampling [\cite{russel2019model}], using the \verb;NS; BEAST2 package. 

If twinning is used, a candidate model that has the highest evidence for
the twin alignment is also used to create the twin error
distribution.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Stochasticity caused by simulating phylogenies}
\label{subsec:stochasticity}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The goal is to evaluate BEAST2's performance 
on a non-standard tree model, 
one must also consider the last source of stochasticity: 
the different phylogenies a tree model generates.
A single phylogeny cannot be considered as fully representative of the model. 
For this reason multiple phylogenies must be 
considered \new{(at least 100 independent true and twin trees)}.
If the number of considered phylogenies is high enough, 
the comparison between the main pipeline's aggregated error distribution 
and its twin counterpart leads to a fair evaluation 
of the new tree model with respect to the baseline error.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The nLTT statistic}
\label{subsec:nltt}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The nLTT statistic is the absolute difference
between the normalized lineages-through-time plots of two trees.
The nLTT statistic is chosen, as it can operate on any two trees (regardless
of their crown ages and number of taxa) and its results have a clear range
from zero to one. This normalized result makes it possible to compare trees 
from a distribution of trees from any tree model.
\new{
  The nLTT statistic is not suitable, however, to distinguish
  between a constant-rate BD model and a family of time-dependent 
  models [\cite{louca2020extant}].
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Main functions}
\label{subsec:main_functions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

An overview of \verb;pirouette;'s main functions is shown in 
Table~\ref{tab:functions}. 
All \verb;pirouette;'s functions are documented,
have a useful example and sensible defaults.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{table}[h]
  \centering
  \begin{tabular}{ | l | l | l | }
    \hline
    \textbf{Name} & \textbf{Description} \\
    \hline
    \verb;pir_run; & Run \verb;pirouette; \\
    \verb;pir_plot; & Show the \verb;pirouette; results as a plot  \\
    \verb;create_pir_params; & Create the \verb;pirouette; parameters  \\
    \hline
    \verb;create_alignment_params; & Create the alignment parameters  \\
    \verb;create_twinning_params; & Create the twinning parameters  \\
    \verb;create_experiment; & Create one experiment  \\
    \verb;create_error_measure_params; & Create the error measurement parameters  \\
    \hline
  \end{tabular}
  \caption{
    \texttt{pirouette}'s main functions and description. 
  }
  \label{tab:functions}
\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Main example}
\label{subsec:main_example}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

This subsection describes the pipeline of the main example 
and its diagnostics in more detail. 

The pipeline starts at the top-left panel of figure \ref{fig:pipeline} (which 
is identical to figure \ref{fig:dd_tree}),
which is the 'true tree'.
The 'true tree' is generated by the diversity-dependent (DD) 
tree model [\citep{DDD, etienne2012diversity}],
which is a BD model with a speciation rate that is dependent on the number of species,
with (an arbitrarily chosen) crown age of 10 time units 
and 6 tips for an extinction rate of 0.1. 
The carrying-capacity is set to 6. 
The initial speciation rate $\lambda_0$ is chosen 
such that the expected number of species in a constant-rate BD model 
would be equal to the number of tips, which amounts to $\lambda_0 = 0.63$.

From this 'true tree', a 'true alignment' is simulated, using
the JC nucleotide substitution model and a strict clock model.
The resulting alignment is shown at the center-left
of figure \ref{fig:pipeline}.

From the 'true alignment' the generative inference model is run.
Of course, it cannot be the actual (DD) model. Instead, the
default BEAST2 inference model is used, which assumes a JC nucleotide
substitution model, a strict clock model and a Yule tree model.
The resulting posterior trees are shown in the
center-left panel of figure \ref{fig:pipeline}.

From this 'generative true' posterior (center-left panel in figure \ref{fig:pipeline}), the difference between each of its trees is
compared to the 'true tree' (top-left panel), using the nLTT statistic,
resulting in the error distribution shown in the bottom-left panel
of figure \ref{fig:pipeline}.

Based on the 'true alignment' (center-left panel), the candidate model
with the highest marginal likelihood is determined, from a set of 
15 models. The set of models consists of all combinations
of all 4 nucleotides substitution models (JC, HKY, TN, GTR),
all 2 clock models (strict and relaxed log-normal) and 2 birth-death
models (Yule and Birth-Death), except the inference model used as the
generative model (JC, strict clock, Yule). The inference model that had the
highest evidence (as shown in Table \ref{tab:evidence_true}) was 
the inference model with a JC nucleotide substitution model,
a relaxed log-normal clock model and a Yule tree model.
The resulting posterior trees are shown in the second panel
of the third row of posteriors in figure \ref{fig:pipeline}.

From this 'best true' posterior, the difference between each of its trees is
compared to the 'true tree' (top-left panel), using the nLTT statistic,
resulting in the second error distribution in the bottom row
of figure \ref{fig:pipeline}.

From the 'true tree' (top-left) we generated a BD twin tree (top-right).

From this 'twin tree', a 'twin alignment' was simulated, using
the JC nucleotide substitution model and a strict clock model.
The resulting alignment is shown in the center-right panel
of figure \ref{fig:pipeline}.

From the 'twin alignment' the generative inference model is run as well.
Also here, the default BEAST2 inference model is used, 
which assumes a JC nucleotide
substitution model, a strict clock model and a Yule tree model.
The resulting posterior trees are shown in the
third panel of the third row of figure \ref{fig:pipeline}.
From this 'generative twin' posterior, 
the difference between each of its trees is
compared to the 'twin tree' (top-right panel), using the nLTT statistic,
resulting in the error distribution shown in the third panel
of the bottom row of figure \ref{fig:pipeline}.

Based on the 'twin alignment' (center-right panel), the candidate model
with the highest marginal likelihood is determined, from the same 
set of 15 candidate models.The inference model that had the
highest evidence (as shown in Table \ref{tab:evidence_twin}) was 
the inference model with a JC nucleotide substitution model,
a strict clock model and a BD tree model (note: this indeed matches how
the twin tree and twin tree were simulated).
From the 'twin alignment' this best candidate inference model is run.
The resulting posterior trees are shown in the fourth panel
of the third row of posteriors in figure \ref{fig:pipeline}.

From this 'best twin' posterior (fourth in third row of
figure \ref{fig:pipeline}), the difference between each of its trees was
compared to the 'twin tree' (top-right panel), using the nLTT statistic,
resulting in the fourth error distribution in the bottom row
of figure \ref{fig:pipeline}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[H]
  \centering
  \resizebox {1.0\columnwidth} {!} {
    \begin{tikzpicture}[
      ->,>=stealth',shorten >=1pt,auto,
      node distance=0.5\textheight, 
      semithick
    ]   
    \tikzstyle{every state}=[]
    \node[state, draw=none] (O) [] {
    };   
    \node[state] (A) [right of = O, rectangle] {
      \includegraphics[height=0.4\textheight]{pirouette_example_30/example_30/true_tree.png}
    };   
    \node[state] (B) [below of = A, rectangle] {
      \includegraphics[height=0.25\textheight]{pirouette_example_30/example_30/true_alignment.png}
    };   
    \node[state] (CG) [below of = B, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_posterior_gen.png}
    };   
    \node[state] (DG) [below of = CG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_error_violin_gen.png}
    };   
    \node[state] (CB) [right of = CG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_posterior_best.png}
    };   
    \node[state] (DB) [below of = CB, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_error_violin_best.png}
    };   
    \node[state] (AT) [right of = A, rectangle, node distance=0.8\textheight] {
      \includegraphics[height=0.4\textheight]{pirouette_example_30/example_30/twin_tree.png}
    };   
    \node[state] (BT) [below of = AT, rectangle] {
      \includegraphics[height=0.25\textheight]{pirouette_example_30/example_30/twin_alignment.png}
    };   
    \node[state] (CTG) [right of = CB, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_posterior_gen.png}
    };   
    \node[state] (DTG) [below of = CTG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_error_violin_gen.png}
    };   
    \node[state] (CTB) [right of = CTG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_posterior_best.png}
    };   
    \node[state] (DTB) [below of = CTB, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_error_violin_best.png}
    };   
    \path 
      (O) edge [anchor = south] node {} (A)
      (A) edge [anchor = south] node {} (B)
      (B) edge [anchor = south] node {} (CG)
      (CG) edge [anchor = south] node {} (DG)
      (B) edge [anchor = south east] node {} (CB)
      (CB) edge [anchor = south] node {} (DB)
      (A) edge [anchor = east] node {} (AT)
      (AT) edge [anchor = south] node {} (BT)
      (BT) edge [anchor = south east] node {} (CTG)
      (CTG) edge [anchor = south] node {} (DTG)
      (BT) edge [anchor = south] node {} (CTB)
      (CTB) edge [anchor = south] node {} (DTB)
    ; 
    \end{tikzpicture}
  }
  \caption{Full pirouette pipeline, including comparison to baseline error. 
    The true tree (top left) is used to simulate an alignment. 
    From this alignment two posterior distributions of trees are created: 
    one using the generative model and another one using the inference model 
    with the highest marginal likelihood. 
    For each distribution of trees, a distribution of errors, 
    measured with the nLTT statistic, 
    between the posterior trees and the main trees is drawn. 
    From the true tree also a twin tree is created (right side of the figure)
    which follows the same pipeline, 
    leading to two additional error distributions to use as baseline errors.}
  \label{fig:example_30_full_pipeline}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

To assess if the results of the inference are meaningful one important 
parameter is the Effective Sample Size (ESS). This quantity describes how 
many independent trees are sampled from the posterior distributions. 
For reliable results it is good practice to have at 
least $ESS = 200$ (see 
\begin{sloppypar}
  \url{https://beast.community/ess_tutorial}).
\end{sloppypar}
In the following we present the ESS for the posterior distributions of 
the 4 cases shown in Fig. \ref{fig:example_30_full_pipeline}.

The ESSes of the 'true' pipeline for the generative model
are shown in Table \ref{tab:esses_gen}.
From the estimated parameters, one can deduce that
the JC nucleotide substitution model was used (no
estimated parameter needed), a strict clock model was 
used (again, no parameter needed to be estimated)
and a Yule tree prior is used ('Yule model' and 'birthRate' are estimated).
Note that although the actual true tree is created by a DD process, 
the default and standard Yule tree model is used as the closest
standard tree model.

\input{pirouette_example_30/example_30/esses_gen.latex}

\newpage

The ESSes of the 'twin' pipeline for the generative model
are shown in Table \ref{tab:esses_twin_gen}.
Note that the generative inference model is 
re-used (which assumes a Yule tree model) in the inference, 
where the twin tree is actually created using a BD process,
which is the default.

\input{pirouette_example_30/example_30/esses_twin_gen.latex}

\newpage

The ESSes of the 'true' pipeline for the best candidate model
are shown in Table \ref{tab:esses_best}. 
From the names of the estimated parameters, it is clear that
the best candidate model has had a TN nucleotide substitution 
model (which can be inferred from the parameters'kappa1', 'kappa2', 'freqParameter.1', 
'freqParameter.2', 'freqParameter.3', 'freqParameter.4')
a strict clock (no parameter needed to be estimated) and a BD 
model ('BirthDeath', 'BDBirthRate' and 'BDDeathRate').

\input{pirouette_example_30/example_30/esses_best.latex}

\newpage

The ESSes of the 'twin' pipeline for the best candidate model
are shown in Table \ref{tab:esses_twin_best}.

From the names of the estimated parameters, it is clear that
the best candidate model for the twin tree
is JC nucleotide substitution 
model (no parameter needed to be estimated),
a strict clock (no parameter needed to be estimated) and a BD 
model ('BirthDate', 'BDBirthRate' and 'BDDeathRate').
Note that this matches the actual process of how the twin tree
and twin alignment are generated.

\input{pirouette_example_30/example_30/esses_twin_best.latex}

\newpage

The marginal likelihood (or evidence) data for the model comparison
performed in the 'true' pipeline is shown in Table \ref{tab:evidence_true}.
The best (that is, the one with the highest model weight)
candidate model assumes a TN nucleotide substitution model,
a strict clock and a BD model.

\input{pirouette_example_30/example_30/evidence_true.latex}

\newpage

The marginal likelihood (or evidence) data for model comparison
performed in the 'twin' pipeline is shown in Table \ref{tab:evidence_twin}.
The best (that is, the one with the highest model weight)
candidate model assumes a JC nucleotide substitution model,
a strict clock and a BD model.
Note that this matches the actual process of how the twin tree
and twin alignment are generated.

\input{pirouette_example_30/example_30/evidence_twin.latex}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Using a distribution of trees}
\label{subsec:distribution}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

This subsection extends the main example, by using multiple (instead of
one) trees. These trees are produced by running a DD tree simulation
with the same parameters as the main example.

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_28/errors.png}
  \caption{
    Aggregate error distributions, 
    similar to Fig. \ref{fig:example_30} for the main example, 
    but now for a collection of 100 replicate trees. 
    For each setting (true generative, true best candidate, 
    twin generative and twin best candidate), the resulting errors 
    from each replicate pipeline have been merged into a single distribution. 
    This took 2.7 days (wall clock time) to compute.
  }
  \label{fig:replicate_trees}
\end{figure}

\new{
  The resulting error distributions are shown in Fig. \ref{fig:replicate_trees}. 
  We present results for cases where (1) the generative model has been used 
  or (2) the model with highest evidence has been selected for the inference.
  From the plots we can see that in both cases the two distributions 
  (true and twin) are mostly overlapping, but not everywhere.
  This suggests that the inference 
  models that have been used can to a reasonable extent capture in an accurate way 
  the features of the diversity-dependent tree prior used to simulate 
  the original trees. 
}

The code to reproduce Fig. \ref{fig:replicate_trees} can be found at  
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_28}.
\end{sloppypar}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of the number of taxa}
\label{subsec:n_taxa}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses 6 taxa. Here we show the same results as the main example,
except for a varying number of taxa. We did so, by setting the DD model's
carrying capacity to the desired number of taxa.

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_32/errors.png}
  \caption{Aggregate error distributions for 100 replicates. Here each true tree has 12 taxa. This took 6.0 days (wall clock time) to compute.}
  \label{fig:example_12_taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_33/errors.png}
  \caption{Aggregate error distributions for 100 replicates. Here each true tree has 24 taxa. This took 9.8 days (wall clock time) to compute.}
  \label{fig:example_24_taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_41/errors.png}
  \caption{Aggregate error distributions for 65 replicates. Here each true tree has 32 taxa. This took 8.0 days (wall clock time) to compute.}
  \label{fig:example_32_taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_42/errors.png}
  \caption{Aggregate error distributions for 5 replicates. Here each true tree has 40 taxa. This took 0.83 days (wall clock time) to compute.}
  \label{fig:example_40_taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{supplementary_figures/plot_error_vs_n_taxa.png}
  \caption{Difference between median true error and median twin error for different number of taxa.}
  \label{fig:error_vs_ntaxa}
\end{figure}

\new{
  We show in figures \ref{fig:replicate_trees}, \ref{fig:example_12_taxa}, 
  \ref{fig:example_24_taxa}, \ref{fig:example_32_taxa} and 
  \ref{fig:example_40_taxa} what are the errors obtained when starting from 
  phylogenies with, respectively, 6, 12, 24, 32 and 40 taxa.
  Again we can see 
  that in each case errors tend to be greater in the true distribution than 
  in the twin distribution, similar to the result of subsection \ref{subsec:distribution}.
  Collecting all the data together we can see that errors tend to decrease as the number of taxa in the considered phylogenies increase (see Fig. \ref{fig:error_vs_ntaxa}). The data point for $40$ taxa not following the trend could be due to the limited amount of simulated trees taken in consideration due to time constraints.
}

The code to reproduce these figures can be found at  
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_28} (6 taxa, main example),
  \url{https://github.com/richelbilderbeek/pirouette_example_32} (12 taxa),
  \url{https://github.com/richelbilderbeek/pirouette_example_33} (24 taxa),
  \url{https://github.com/richelbilderbeek/pirouette_example_41} (32 taxa),
  \url{https://github.com/richelbilderbeek/pirouette_example_42} (40 taxa). 
\end{sloppypar}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of DNA sequence length}
\label{subsec:n_nucleotides}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a DNA alignment length of 1000 nucleotides.
Here, we show the same results as the main example,
except for a varying DNA alignment sequence length.

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_19/errors.png}
  \caption{Aggregate error distributions for 100 replicates. 
    Here each each alignment has a sequence length of 500 nucleotides. 
    This took 2.2 days (wall clock time) to compute.}
  \label{fig:example_500_nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_28/errors.png}
  \caption{Aggregate error distributions for 100 replicates. 
    Here each each alignment has a sequence length of 1000 nucleotides.
    This is a replicate of Fig. \ref{fig:replicate_trees}. We put it here to facilitate the comparison with the cases with different number of nucleotides.}
  \label{fig:example_1000_nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_34/errors.png}
  \caption{Aggregate error distributions for 100 replicates. 
    Here each each alignment has a sequence length of 2000 nucleotides. 
    This took 4.4 days (wall clock time) to compute.}
  \label{fig:example_2000_nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{supplementary_figures/plot_error_vs_sequence_length.png}
  \caption{Difference between median true error and median twin error for different sequence lengths.}
  \label{fig:error_vs_seqlength}
\end{figure}

\new{
  From figures \ref{fig:example_500_nucleotides}, 
  \ref{fig:example_1000_nucleotides} and \ref{fig:example_2000_nucleotides} 
  we can observe that the discrepancy between the true and twin error 
  distributions tends to become smaller as the number of nucleotides increase (see also Fig. \ref{fig:error_vs_seqlength}).
  This occurred for both the generative and best candidate cases.
  This follows the expectation that a prior becomes less important when
  more information becomes available.
}

The code to reproduce these figures can be found at  
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_19} (500 nucleotides),
  \url{https://github.com/richelbilderbeek/pirouette_example_28} (1000 nucleotides, main example),
  and \url{https://github.com/richelbilderbeek/pirouette_example_34} (2000 nucleotides).
\end{sloppypar}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of assuming a Yule tree prior on a Yule tree}
\label{subsec:simplest_correct_parameterization}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a tree generated by a non-standard tree model.
Here, we show the same results, with the only difference that
the tree used is generated by simplest tree model (the Yule model),
which we also assume as the (correct) tree prior.

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_22/errors.png}
  \caption{Aggregate error distributions for 100 replicates. 
    Here each true tree is generated by a Yule process. 
    For the inference we used a Yule tree prior. 
    This took 2.9 days (wall clock time) to compute.}
  \label{fig:example_yule}
\end{figure}

This example shows a parameterization at the correct level for the
simplest case possible.

\new{
  As expected the twin and true distributions in Fig. \ref{fig:example_yule} 
  are extremely similar for both the generative and the best candidate case.
}

The code to reproduce this figure can be found at  
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_22}.
\end{sloppypar}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of assuming a Yule tree prior on a BD tree}
\label{subsec:under_parameterization}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a tree generated by a non-standard tree model.
Here, we show the same results, with the difference that
the tree used is generated by a birth-death (BD) tree model,
where we assume it is generated by a Yule (or pure-birth) model.
This example thus shows the effect of underparameterization.

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_26/errors.png}
  \caption{Aggregate error distributions for 100 replicates. 
    Here each true tree is generated by a BD process. 
    For the inference we used instead a Yule tree prior. 
    This took 2.7 days (wall clock time) to compute.}
\end{figure}

\new{
  Because the two models are very similar to each other (the BD model can be 
  turned into a Yule model just by setting the extinction parameter to 
  zero [\cite{nee1994reconstructed}]) the median discrepancy is
  almost negligible. 
  However, with respect to the previous 
  case (subsection \ref{subsec:simplest_correct_parameterization}), 
  where a Yule tree prior was used, 
  the distributions here exhibit a greater difference.
}
\new{As we use only extant trees, it is reasonable that the method
  is slightly weaker in distinguishing between the Yule and BD
  models. It is unknown what the discriminatory power would be
  when comparing trees with extinction events.}

The code to reproduce this figure can be found at
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_26}.
\end{sloppypar}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of diversity-dependent trees differing in how likely they are under the DD process}
\label{subsec:better_label_needed}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Here we show the results of a \verb;pirouette; run on a dataset 
of multiple DD trees that we selected for having a low, median
and high likelihood. In this way, we effectively selected for trees
that are rare, uncommon and common respectively.

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_23/errors_low.png}
  \caption{Aggregate error distributions for a distribution of trees, where the true trees are DD with low likelihood.}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_23/errors_mid.png}
  \caption{Aggregate error distributions for a distribution of trees, 
    where the true trees are DD with median likelihood.}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_23/errors_high.png}
  \caption{
    Aggregate error distributions for a distribution of trees, 
    where the true trees are DD with high likelihood.
  }
\end{figure}

\new{
  Here the median errors are similar in the three settings and similar to 
  ones relative to the full dataset of \ref{subsec:distribution}. We can also 
  notice that in the case of median likelihood, the twin median error appears 
  to be lower than the true mean error. This is usually a sign that the number 
  of replicates (in this case 10) is too low to allow us to draw precise 
  conclusions from this test. We did not explore further in this 
  direction using more simulations because computational times turned to be 
  extremely high. 
  The entire run took 120 hours in total.
}

The code to reproduce these figure can be found at  
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_23}
\end{sloppypar}.

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of equal or equalized mutation rate in the twin alignment}
\label{subsec:different_n_mutations}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  
The main example uses a twin alignment that has the same number
of \new{substitution}s (as measured from the ancestral sequence) as the true alignment. Here, we show the same results, with the difference that
the twin alignment uses the same mutation rate, yet is not guaranteed
to have the same number of \new{substitution}s.

\begin{figure}[H]
  %\includegraphics[width=0.98\textwidth]{pirouette_example_18/errors.png}
  \includegraphics[width=0.94\textwidth]{pirouette_example_18/errors.png}
  \caption{Aggregate error distributions for 100 replicates.
    similar to Fig. \ref{fig:replicate_trees}, 
    but here the number of \new{substitution}s is not imposed 
    to be the same between true and twin alignment. 
    Instead, an equal mutation rate is used. 
    This took 3.3 days (wall clock time) to compute.}
  \label{fig:example_random_mutations}
\end{figure}

\new{
  Comparing figures \ref{fig:example_random_mutations} 
  and \ref{fig:replicate_trees} we can see that the discrepancy between 
  true and twin distributions tend to increase. 
  This is probably due to the fact that letting mutation rates 
  induces a difference in the amount of information contained in the 
  alignments and this is reflected in the error distributions.
}

The code to reproduce this figure can be found at
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_18} and
  \url{https://github.com/richelbilderbeek/pirouette_example_28}.
\end{sloppypar}

\newpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of mutation rate}
\label{subsec:mutation_rate}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a mutation rate such that all nucleotides,
on average, mutate once over the history going from the
ancestral sequence at the crown to the alignments at the tips.
This value equals `1.0 / crown age`.
In this way, the alignment is expected to contain the maximum
amount of information.

Here, we show the same results for different mutation rates.
\new{
\iffalse
  We can observe that the error tends to increases for mutation rates above
  `1.0 / crown age`. Below this value, the errors are the same.
  For these lower mutation rates, however, the shape of the error
  distribution changes for the baseline/generative models,
  deviating more and more from the gamma-distribution-like shape
  as is still present in the twin distributions.
\fi
 The results for the different mutation rates are shown in Figs. \ref{fig:example_0.25_mutation_rate} (0.25 / crown age), \ref{fig:example_0.50_mutation_rate} (0.50 / crown age), \ref{fig:example_0.75_mutation_rate} (0.75 / crown age), \ref{fig:replicate_trees} (1.00 / crown age), \ref{fig:example_1.25_mutation_rate} (1.25 / crown age),
 \ref{fig:example_1.50_mutation_rate} (1.50 / crown age) and
 \ref{fig:example_2.00_mutation_rate} (2.00 / crown age).
 }
\new{
 Fig. \ref{fig:error_vs_mutationrate} summarizes all the other figures showing on the y-axis, for each value of the mutation rate, the difference between the median of the true distribution and the median of the twin distribution. We can observe a general positive trend as the mutation rate increase, even though the value for 1.5 / crown age suggests to take this result with caution. It is possible, however, that a more regular trend could be observed increasing the number of simulations.
}

The code to reproduce this figure can be found at
\begin{sloppypar}
  \url{https://github.com/richelbilderbeek/pirouette_example_35} (0.25 / crown age),
  \url{https://github.com/richelbilderbeek/pirouette_example_36} (0.50 / crown age),
  \url{https://github.com/richelbilderbeek/pirouette_example_37} (0.75 / crown age),
  \url{https://github.com/richelbilderbeek/pirouette_example_28} (1.00 / crown age, example reported in \ref{subsec:distribution}, see Fig. \ref{fig:replicate_trees}),
  \url{https://github.com/richelbilderbeek/pirouette_example_38} (1.25 / crown age),
  \url{https://github.com/richelbilderbeek/pirouette_example_39} (1.50 / crown age),
  \url{https://github.com/richelbilderbeek/pirouette_example_40} (2.00 / crown age),
\end{sloppypar}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_35/errors.png}
  \caption{Aggregate error distributions for 100 replicates,
    for the tree distribution presented 
    in \ref{subsec:distribution} but with a per-nucleotide mutation rate 
    of 0.25 / crown age. 
    This took 1.8 days (wall clock time) to compute.}
  \label{fig:example_0.25_mutation_rate}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_36/errors.png}
  \caption{Aggregate error distributions for 100 replicates,
    for the tree distribution presented 
    in \ref{subsec:distribution} but with a per-nucleotide mutation rate 
    of 0.50 / crown age. 
    This took 2.2 days (wall clock time) to compute.}
  \label{fig:example_0.50_mutation_rate}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_37/errors.png}
  \caption{Aggregate error distributions for 100 replicates,
    for the tree distribution presented 
    in \ref{subsec:distribution} but with a per-nucleotide mutation rate 
    of 0.75 / crown age. 
    This took 2.5 days (wall clock time) to compute.}
  \label{fig:example_0.75_mutation_rate}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_38/errors.png}
  \caption{Aggregate error distributions for 100 replicates,
    for the tree distribution presented 
    in \ref{subsec:distribution} but with a per-nucleotide mutation rate 
    of 1.25 / crown age. 
    This took 2.9 days (wall clock time) to compute.}
  \label{fig:example_1.25_mutation_rate}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_39/errors.png}
  \caption{Aggregate error distributions for 100 replicates,
    for the tree distribution presented 
    in \ref{subsec:distribution} but with a per-nucleotide mutation rate 
    of 1.50 / crown age. 
    This took 3.0 days (wall clock time) to compute.}
  \label{fig:example_1.50_mutation_rate}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{pirouette_example_40/errors.png}
  \caption{Aggregate error distributions for 100 replicates,
    for the tree distribution presented 
    in \ref{subsec:distribution} but with a per-nucleotide mutation rate 
    of 2.0 / crown age. 
    This is done for 100 replicates.
    This took 3.0 days (wall clock time) to compute.}
  \label{fig:example_2.00_mutation_rate}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=0.98\textwidth]{supplementary_figures/plot_error_vs_mutation_rate.png}
  \caption{Difference between median true error and median twin error for different values of mutation rate.}
  \label{fig:error_vs_mutationrate}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Acknowledgments}
\label{subsec:acknowledgments}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

We thank the Center for Information Technology of the University 
of Groningen for its support and for providing access to the Peregrine 
high performance computing cluster. 
We thank the Netherlands 
Organization for Scientific Research (NWO) for financial support 
through a VICI grant awarded to RSE.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Data accessibility}
\label{subsec:data_accessibility}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

All code is archived at 
\url{http://github.com/richelbilderbeek/pirouette_article},
with DOI \url{https://doi.org/12.3456/zenodo.1234567}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Author contributions}
\label{subsec:author_contributions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

RJCB, GL and RSE conceived the idea for the package. 
RJCB created, tested and revised the package.
GL provided major contributions to the package.
RJCB wrote the first draft of the manuscript, 
GL and RSE contributed to revisions.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Bibliography
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Keep the next commented LaTeX commands
%
% These two lines must be added in a thesis, 
%   so that the references have a proper link.
% This is done by sed in the Makefile of github.com/richelbilderbeek/thesis
%
%\bibliographystyle{pirouette_mee}
%\bibliography{pirouette_article}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

