%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Supplementary materials}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

These supplementary materials show the full pipeline and
added diagnostics for the examples in the main article.

All the figures shown in this section are shown as-is, 
without any aesthetical modifications, with the exception that the arrangement
of the sub-figures (for example, aligning true and twin tree
horizontally) is done manually.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Overview}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\itemize{
  \item{
    subsection \ref{subsec:main_example}: extra figures and
    diagnostics regarding the main example.
  }
  \item{
    subsection \ref{subsec:distribution}: the result of using 
    multiple trees, as generated by the
    same stochastic process as the main example
  }
  \item{
    subsection \ref{subsec:n_taxa}: the effect of the number
    of taxa
  }
  \item{
    subsection \ref{subsec:n_nucleotides}: the effect of the DNA
    alignment sequence length
  }
  \item{
    subsection \ref{subsec:non_clock}: the effect when the alignment
    is generated by a non-clock like nucleotide substitution model
  }
  \item{
    subsection \ref{subsec:simplest_correct_parameterization} shows the
    effect when doing an inference in the simplest use case
  }
  \item{
    subsection \ref{subsec:under_parameterization} shows the
    effect when doing an inference with an under-parameterization
  }
  \item{
    subsection \ref{subsec:different_n_mutations} shows the
    effect when the twin alignment is allowed to have a different
    number of mutations
  }
  \item{
    subsection \ref{subsec:mutation_rate} shows the
    effect of different mutation rates
  }
  \item{
    subsection \ref{subsec:better_label_needed} shows 
    \richel{better description needed}
  }
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Main example}
\label{subsec:main_example}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

This subsection shows the diagnostics from the main example, which
uses one tree.
The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_30}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[H]
  \centering
  \resizebox {1.0\columnwidth} {!} {
    \begin{tikzpicture}[
      ->,>=stealth',shorten >=1pt,auto,
      node distance=0.5\textheight, 
      semithick
    ]   
    \tikzstyle{every state}=[]
    \node[state, draw=none] (O) [] {
    };   
    \node[state] (A) [right of = O, rectangle] {
      \includegraphics[height=0.4\textheight]{pirouette_example_30/example_30/true_tree.png}
    };   
    \node[state] (B) [below of = A, rectangle] {
      \includegraphics[height=0.25\textheight]{pirouette_example_30/example_30/true_alignment.png}
    };   
    \node[state] (CG) [below of = B, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_posterior_gen.png}
    };   
    \node[state] (DG) [below of = CG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_error_violin_gen.png}
    };   
    \node[state] (CB) [right of = CG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_posterior_best.png}
    };   
    \node[state] (DB) [below of = CB, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/true_error_violin_best.png}
    };   
    \node[state] (AT) [right of = A, rectangle, node distance=0.8\textheight] {
      \includegraphics[height=0.4\textheight]{pirouette_example_30/example_30/twin_tree.png}
    };   
    \node[state] (BT) [below of = AT, rectangle] {
      \includegraphics[height=0.25\textheight]{pirouette_example_30/example_30/twin_alignment.png}
    };   
    \node[state] (CTG) [right of = CB, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_posterior_gen.png}
    };   
    \node[state] (DTG) [below of = CTG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_error_violin_gen.png}
    };   
    \node[state] (CTB) [right of = CTG, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_posterior_best.png}
    };   
    \node[state] (DTB) [below of = CTB, rectangle] {
      \includegraphics[height=0.3\textheight]{pirouette_example_30/example_30/twin_error_violin_best.png}
    };   
    \path 
      (O) edge [anchor = south] node {} (A)
      (A) edge [anchor = south] node {} (B)
      (B) edge [anchor = south] node {} (CG)
      (CG) edge [anchor = south] node {} (DG)
      (B) edge [anchor = south east] node {} (CB)
      (CB) edge [anchor = south] node {} (DB)
      (A) edge [anchor = east] node {} (AT)
      (AT) edge [anchor = south] node {} (BT)
      (BT) edge [anchor = south east] node {} (CTG)
      (CTG) edge [anchor = south] node {} (DTG)
      (BT) edge [anchor = south] node {} (CTB)
      (CTB) edge [anchor = south] node {} (DTB)
    ; 
    \end{tikzpicture}
  }
  \label{fig:example_30_full_pipeline}
  \caption{Comparing to background noise: full pipeline}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\input{pirouette_example_30/example_30/esses_gen.latex}

\input{pirouette_example_30/example_30/esses_best.latex}

\input{pirouette_example_30/example_30/esses_twin_gen.latex}

\input{pirouette_example_30/example_30/esses_twin_best.latex}

\input{pirouette_example_30/example_30/evidence_true.latex}

\input{pirouette_example_30/example_30/evidence_twin.latex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Using a distribution of trees}
\label{subsec:distribution}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\richel{Issue \url{https://github.com/richelbilderbeek/pirouette_article/issues/89}}

This subsection extends the main example, by using multiple (instead of
one) trees. These trees are produced by running a DD tree simulation
with the same parameters as the main example, which are
speciation rate \richel{?}, extinction rate $0.1$, and carrying
capacity 40. 

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_28}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_28/errors.png}
  \caption{Error distribution from a distribution of trees}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of the number of taxa}
\label{subsec:n_taxa}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\richel{Issue at \url{https://github.com/richelbilderbeek/pirouette_article/issues/63}}

The main example uses \richel{?} taxa.
One expects that the more taxa are present in a phylogeny,
the better the inference will be, 
thus the lower the errors will be.
Here, we show the same results as the main example,
except for a varying number of taxa.

Note that, \richel{[some authors]} found that an increase in the number
of taxa may counterintuitively decrease the inference.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_20}. 


\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_20/errors_5.png}
  \caption{5 taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_20/errors_10.png}
  \caption{10 taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_20/errors_20.png}
  \caption{20 taxa}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_20/errors_30.png}
  \caption{30 taxa}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of DNA sequence length}
\label{subsec:n_nucleotides}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a DNA alignment length of 1000 nucleotides.
One expects that the more nucleotides are present in the alignment,
the more information is available,
thus the better the inference will be, 
thus the lower the errors will be.
Here, we show the same results as the main example,
except for a varying DNA alignment sequence length.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_21}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_100.png}
  \caption{100 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_248.png}
  \caption{248 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_500.png}
  \caption{500 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_1000.png}
  \caption{1000 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_2000.png}
  \caption{2000 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_4000.png}
  \caption{4000 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_8000.png}
  \caption{8000 nucleotides}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_21/errors_16000.png}
  \caption{16000 nucleotides}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of non-clock like models}
\label{subsec:non_clock}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\richel{Issue at \url{https://github.com/richelbilderbeek/pirouette_article/issues/87}}

The main example generates an alignment using a strict clock model (that is,
all species have an equal mutation rate).
One may wonder, how well the inference will be for a different non-clock like
model.
Here, we show the same results as the main example,
except that instead of generating the alignment with a strict clock
model, we use an unlinked node substitution model (by Thijs Janzen,
unpublished, code available at \url{https://github.com/thijsjanzen/nodeSub}),
with a branch mutation rate of $0.1$ and a node mutation rate of $0.1$.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_27}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_27/errors.png}
  \caption{An unliked node substitution model}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of assuming a Yule tree prior on a Yule tree}
\label{subsec:simplest_correct_parameterization}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a tree generated by a non-standard tree model.
Here, we show the same results, with the only difference that
the tree used is generated by simplest tree model (the Yule model),
which we also assume as the (correct) tree prior.
This example thus shows a parameterization at the correct level for the
simplest case possible.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_22}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_22/errors.png}
  \caption{Assuming a Yule tree prior on a Yule tree}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of assuming a Yule tree prior on a BD tree}
\label{subsec:under_parameterization}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The main example uses a tree generated by a non-standard tree model.

Here, we show the same results, with the difference that
the tree used is generated by a birth-death (BD) tree model,
where we assume it is generated by a Yule (or pure-birth) model.
This example thus shows the effect of underparameterization.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_26}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_26/errors.png}
  \caption{Assuming a Yule tree prior on a BD tree}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of differently common diversity-dependent trees}
\label{subsec:better_label_needed}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\richel{Issue at \url{https://github.com/richelbilderbeek/pirouette_article/issues/68}}

Here we show the results of a \verb;pirouette; run on a dataset 
of $N = 100$ \giovanni{N can be changed if 100 is too much} 
trees simulated under a diversity-dependent process. 
All the trees are simulated using the DDD package (\cite{DDD}). 
Our goal is to show how trees simulated by a process that is 
sensibly different from a standard tree prior will result in a 
significant divergence between true and twin distribution. 
To do so we set the simulation parameters (in this case: 
absolute speciation rate $\lambda_0 = 1.6$, 
extinction rate $\mu = 0.2$ and carrying capacity $K = 20$) 
in order to express clear characteristics of diversity dependence in the 
phylogenies. In fact, for a high speciation rate and a low carrying capacity, 
we expect trees to exhibit high unbalance, compared with trees generated 
by a standard birth-death model. 
We can measure how unbalanced is a tree using the $\gamma$ 
statistics (\cite{pybus2000testing}). For this reason, we choose to 
measure the error made by the inference process using the $\gamma$ error statistics, 
which is currently implemented in \verb;pirouette;.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_23}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_23/example_23/314/errors.png}
  \caption{Lowest likelihood}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_23/example_23/315/errors.png}
  \caption{Between lowest and median likelihood}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_23/example_23/316/errors.png}
  \caption{Median likelihood}
\end{figure}


\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_23/example_23/317/errors.png}
  \caption{Between median and highest likelihood}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_23/example_23/318/errors.png}
  \caption{Highest likelihood}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of equal or equalized mutation rate in the twin alignment}
\label{subsec:different_n_mutations}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\richel{Issue at \url{https://github.com/richelbilderbeek/pirouette_article/issues/69}}
  
The main example uses a twin alignment that has the same number
of mutations (as measured from the ancestral sequence) as the true alignment.

Here, we show the same results, with the difference that
the twin alignment uses the same mutation rate, yet is not guaranteed
to have the same number of mutations.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_18}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_18/example_18/314/errors.png}
  \caption{Equal mutation rate}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{The effect of mutation rate}
\label{subsec:mutation_rate}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\richel{Issue at \url{https://github.com/richelbilderbeek/pirouette_article/issues/70}}

The main example uses a mutation rate such that all nucleotides,
on average, mutate once over the history going from the
ancestral sequence at the crown to the alignments at the tips.
In this way, the alignment is expected to contain the maximum
amount of information.

Here, we show the same results, with the only difference that
the mutations rates are changed.
For a lower mutation rate, less mutations will be observed, thus
genetic information will be less, thus the inference is expected
to go worse, thus the errors are expected to increase.
For a lower mutation rate, more double-mutations will occur,
scrambling the alignments, resulting in less genetic information, 
thus the inference is expected to go worse, 
thus the errors are expected to increase.

The code used in this part of the article can be found at 
\url{https://github.com/richelbilderbeek/pirouette_example_24}.

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/314/errors.png}
  \caption{Per-nucleotide mutation rate of 0.0125}
\end{figure}


\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/315/errors.png}
  \caption{Per-nucleotide mutation rate of 0.025}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/316/errors.png}
  \caption{Per-nucleotide mutation rate of 0.05}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/317/errors.png}
  \caption{Per-nucleotide mutation rate of 0.1}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/318/errors.png}
  \caption{Per-nucleotide mutation rate of 0.2}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/319/errors.png}
  \caption{Per-nucleotide mutation rate of 0.4}
\end{figure}

\begin{figure}[H]
  \includegraphics[width=\textwidth]{pirouette_example_24/example_24/320/errors.png}
  \caption{Per-nucleotide mutation rate of 0.8}
\end{figure}

